python/ambassador/envoy/v3/v3ready.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2021 Datawire. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+import os
+from typing import TYPE_CHECKING, List
+
+from ...utils import parse_bool
+from .v3listener import V3Listener
+
+if TYPE_CHECKING:
+    from . import V3Config  # pragma: no cover
+
+# The defaults can be changed by using those 2 env vars:
+# AMBASSADOR_READY_PORT: Port number (default 8006)
+# AMBASSADOR_READY_LOG: true/false (default true)
+ambassador_ready_port = int(os.getenv("AMBASSADOR_READY_PORT", "8006"))
+if ambassador_ready_port not in range(1, 32767):
+    ambassador_ready_port = 8006
+ambassador_ready_log = parse_bool(os.getenv("AMBASSADOR_READY_LOG", "true"))
+ambassador_ready_ip = "127.0.0.1"
```

Comment:
Just some clarifying questions:

1. The default Listeners are listening on `0.0.0.0` and this is listening on loopback (127.0.0.1). This was a conscious decision to ensure it is only available for internal implementation use, correct?
2. How does `127.0.0.1` work in an IPV6 only scenario? 


On **2**, we introduced the ability to change the public facing health check bind address, port and IP family in this commit: https://github.com/emissary-ingress/emissary/commit/507bf91f95e489d7d5cb437ca7cd1d8f320d4a81 to support IPV6 only networks. Maybe we need to check the `AMBASSADOR_HEALTHCHECK_IP_FAMILY` and adjust the ready ip to be `::1` for ipv6?

Reply:
I believe that using 127.0.0.1 is deliberate, but: it won't work on an IPv6-only host. That's a great catch: we'll need to switch the IP address based on family, or see if we can use `localhost` here (I suspect not, but worth checking).

Reply:
Good catch. Made a change to use ::1 if AMBASSADOR_HEALTHCHECK_IP_FAMILY is IPV6_ONLY, and 127.0.0.1 in all other cases.

To test this I deployed emissary to an IPv6 kind cluster on my laptop (https://kind.sigs.k8s.io/docs/user/configuration/#networking). First I deployed emissary using the 127.0.0.1 ready address first just to check what kind of issues it causes. For some reason it was working fine without any problems despite the cluster being ipv6 based. 

Afterwards I tried "localhost". Emissary ended up crash looping with this https://gist.github.com/tomasbanet/439fd5dc6d6ea91f9f3171753ea63aea `msg="Post \"http://localhost:8004/_internal/v0/watt?url=http%3A%2F%2Flocalhost%3A9696%2Fsnapshot\": dial tcp [::1]:8004: connect: connection refused" func=github.com/emissary-ingress/emissary/v3/cmd/entrypoint.notifyWebhookUrl file="/go/cmd/entrypoint/notify.go:124"`.

I switched to using either an IPv6 address or IPv4 loopback address explicitly based on the ip family env var. This works for ipv4 but doesn't seem to work when specifying the env var to use ipv6 - https://gist.github.com/tomasbanet/09bcf9659c4ce33b01bfaa6185d2e6e9 `time="2023-01-16 21:02:08.0044" level=error msg="shut down with error error: listen tcp6: address 0.0.0.0: no suitable address found" func=github.com/emissary-ingress/emissary/v3/pkg/busy.Main file="/go/pkg/busy/busy.go:87" CMD=entrypoint PID=1`. I tried deploying emissary from master (without my changes) and it seems to be crashing in the same way. Maybe it's a problem with my cluster set up.

Reply:
@tomasbanet - Thanks for doing thorough testing on this. As of today, we don't explicitly state support for IPV6_ONLY environments but I think it is important we start making moves towards that. Most importantly, I just want to make sure we don't break the recently added health check feature.

@TimonOmsk - I believe you said that you were running Emissary-ingress in IPV6_ONLY mode, is that correct? Would you be willing to check this PR out and make sure it doesn't break you as you? 

Assuming that it doesn't break anything, then I think we can get this merged in here! Thanks for sticking with this.